nifi-registry/nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/db/DatabaseMetadataService.java

```diff
@@ -16,7 +16,21 @@
  */
 package org.apache.nifi.registry.db;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import oracle.jdbc.datasource.OracleCommonDataSource;
```

Comment:
Referencing this specific class is a problem for compatibility with other vendors, so a different approach is necessary to check for Oracle as the version.